pallets/cash/src/core.rs

```diff
@@ -240,6 +243,14 @@ pub fn recover_validator<T: Config>(
                 }
             }
         }
+        ChainSignature::Matic(eth_sig) => {
```

Comment:
will validator only really produce ChainSignature::Eth? do we really need to add more chains here?

Reply:
@waynenilsen this most likely should go, here is the discussion https://discord.com/channels/402910780124561410/796451735803002900/857390332396568596

Reply:
when i took another look at this, i completely agree with you, i _think_ I had to put this in there due to the mechanics of the rest of the code (for now) but it was really not what i wanted to do. I'm going to check out what happens if I take it back out but this is a great call out 

I think the other side of the link is here 

`chains.rs:353`

```
pub fn has_validator_signature(&self, chain_id: ChainId, validator: &ValidatorKeys) -> bool {
        match chain_id {
            ChainId::Eth => self.has_signer(ChainAccount::Eth(validator.eth_address)),
            ChainId::Matic => self.has_signer(ChainAccount::Matic(validator.eth_address)),
            _ => false,
        }
    }
```

need to give this some more thought though for sure, it is something that I struggled with in this PR

Reply:
also this and various other places 

`chains.rs:361`

```
    pub fn add_validator_signature(
        &mut self,
        signature: &ChainSignature,
        validator: &ValidatorKeys,
    ) -> Result<(), Reason> {
        match (self, signature) {
            (ChainSignatureList::Eth(eth_sig_list), ChainSignature::Eth(eth_sig)) => {
                Ok(eth_sig_list.push((validator.eth_address, eth_sig.clone())))
            }
            (ChainSignatureList::Matic(eth_sig_list), ChainSignature::Matic(eth_sig)) => {
                Ok(eth_sig_list.push((validator.eth_address, eth_sig.clone())))
            }
            _ => Err(Reason::SignatureMismatch),
        }
    }
```

definitely need to think about how we want to deal with this.

Basically, right now, i'm just re-using the same keys for eth and matic since their curve and signing situation is compatible `eth_address` may be something of a misnomer, maybe it should be something more like `eth_like_address` or `evm_compatible_address` etc